upstairs/src/volume.rs

```diff
@@ -3824,44 +4042,79 @@ mod test {
         .is_err());
     }
 
-    #[tokio::test]
-    async fn volume_replace_mismatch_vid() {
-        // A replacement VCR is provided with one target being
-        // different, but with incorrect volume layer volume id
+    #[test]
+    // A replacement VCR is provided with one target being different, but with
+    // incorrect volume layer volume id
+    // Test with 1, 2, and 3 sub_volumes.
+    // Test both with each sub_volume having the difference.
+    fn volume_replace_mismatch_vid() {
+        for sv in 1..4 {
+            for sv_changed in 0..sv {
+                volume_replace_mismatch_vid_inner(sv, sv_changed);
+            }
+        }
+    }
+
+    fn volume_replace_mismatch_vid_inner(sv_count: usize, sv_changed: usize) {
         let block_size = 512;
         let vol_id = Uuid::new_v4();
         let blocks_per_extent = 10;
         let extent_count = 9;
 
-        let mut opts = generic_crucible_opts(vol_id);
-        let original = VolumeConstructionRequest::Volume {
-            id: vol_id,
-            block_size,
-            sub_volumes: vec![VolumeConstructionRequest::Region {
+        let opts = generic_crucible_opts(vol_id);
+
+        let mut sub_volumes = Vec::new();
+        for _ in 0..sv_count {
```

Comment:
You know, we should probably just have a `build_subvolume_vcr` helper function instead of copy-pasting this into every single function...

Reply:
Done in ab9e0223